_books/ion-1-1/src/glossary.md

```diff
@@ -0,0 +1,160 @@
+In Ion 1.0 there are two directives: _Ion version markers_, and the _symbol table directives_.
+Ion 1.1 adds _encoding directives_.
+
+**document**<br/>
+A stream of octets conforming to either the Ion text or binary specification.
+Can consist of multiple _segments_, perhaps using varying versions of the Ion specification.
+A document does not necessarily exist as a file, and is not necessarily finite.
+
+**E-expression**<br/>
+See _encoding expression_.
+
+**encoding directive**<br/>
+In an Ion 1.1 segment, a top-level S-Expression annotated with `$ion_encoding`.
+Defines a new encoding module for the segment immediately following it.
+At the end of the encoding directive, the new _encoding module_ is promoted to be the _active encoding module_.
+The _symbol table directive_ is effectively a less capable alternative syntax.
+
+**encoding environment**<br/>
+The context-specific data maintained by an Ion implementation while encoding or decoding data. In
+Ion 1.0 this consists of the current symbol table; in Ion 1.1 this is expanded to also include the Ion
+spec version, the current macro table, and a collection of available modules.
+
+**encoding expression**<br/>
+The invocation of a macro in encoded data, aka e-expression.
+Starts with a macro reference denoting the function to invoke.
+The Ion text format uses "smile syntax" `(:macro ...)` to denote e-expressions. 
+Ion binary devotes a large number of opcodes to e-expressions, so they can be compact.
+
+**encoding module**<br/>
+A _module_ whose symbol table and macro table can be used directly in the user data stream.
+
+**expression**<br/>
+A serialized syntax element that may produce values.
+_Encoding expressions_ and values are both considered expressions, whereas NOP, comments, and IVMs, for example, are not. 
+
+**expression group**<br/>
+A grouping of zero or more _expressions_ that together form one _argument_.
+The concrete syntax for passing a stream of expressions to a macro parameter.
+In a text _e-expression_, a group starts with the trigraph `(::` and ends with `)`, similar to an S-expression.
+In _template definition language_, a group is written as an S-expression starting with `..` (two dots).
+
+**inner module**<br/>
+A _module_ that is defined inside another module and only visible inside the definition of that module.
+
+**Ion version marker**<br/>
+A keyword directive that denotes the start of a new segment encoded with a specific Ion version.
+Also known as "IVM".
+
+**macro**<br/>
+A transformation function that accepts some number of streams of values, and produces a stream of values.
+
+**macro definition**<br/>
+Specifies a macro in terms of a _signature_ and a _template_.
+
+**macro reference**<br/>
+Identifies a macro for invocation or exporting. Must always be unambiguous. Lexically
+scoped, and never a "forward reference" to a macro that is declared later in the document.
```

Comment:
Minor tweak to avoid suggesting that there might be another way to achieve forward referencing.
```suggestion
scoped. Cannot be a "forward reference" to a macro that is declared later in the document;
these are not legal.
```